### PR TITLE
boot: zephyr: Do not use `irq_lock()` if using arm cleanup

### PR DESCRIPTION
**Pull request by sigvartmh**
`irq_lock()` sets `BASEPRI_MAX` aka. the `BASEPRI` mask to whatever
zephyr has configured it to be by the value of `_EXC_IRQ_DEFAULT_PRIO`.

However by calling arm_cleanup() we also do the call to
`__disable_irq()` setting the PRIMASK to 1. Meaning the only exceptions
we can recive is fault exceptions. Masking out more exceptions does not
really make sense.

The rational behind this change is that sometimes applications booted by MCUBoot will not expect the `BASEPRI`
to be set to something other than 0(No effect). Meaning if they depend on
using some exception which now is masked out by `BASEPRI` they will
fail. 

Signed-off-by: Sigvart Hovland <sigvart.hovland@nordicsemi.no>

### CODE REVIEW

**Comment by sigvartmh** [comment]
Another solution could be to `__set_BASEPRI(0)` before jumping to the next image or
```C
unsigned int key = irq_lock()
...
irq_unlock(key);
<jump>
```
Both these should be safe as the only exceptions which are still enabled are fault exceptions.
We could do `__disable_fault_irq ` also if we want to make sure all interrupts except `NMI` does not go through.

**Comment by sigvartmh** [comment]
@ioannisg @d3zd3z @nvlsianpu could someone add someone to review this or review this if you are able to.

**Comment by ioannisg** [comment]
I will take  a look, thanks.

**Review by d3zd3z — APPROVED** [review]
I guess the only real concern I have is whether we will get interrupts before they are expected in the application. Typically, the interrupts we would see in the bootloader would be about the UART, which tend to be relatively slow (and most setups aren't sending characters to the device at this point anyway).

But, I'm ok changing this, and we can deal with any issues that come up.

File: boot/zephyr/main.c [hunk 1]
```diff
@@ -202,6 +201,8 @@ static void do_boot(struct boot_rsp *rsp)
 
 #if CONFIG_CPU_HAS_ARM_MPU || CONFIG_CPU_HAS_NXP_MPU
     z_arm_clear_arm_mpu_config();
+#else
+    irq_lock();
```
Comment (ioannisg) [inline boot/zephyr/main.c:205]: So the irq_lock() is not invoked when the CPU has MPU. Why? What is the relation between cleaning the IRQ state and whether the SoC has the MPU peripheral? I think these two are unrelated.

**Comment by ioannisg** [comment]
> The rational behind this change is that sometimes applications booted by MCUBoot will not expect the BASEPRI
to be set to something other than 0(No effect). Meaning if they depend on
using some exception which now is masked out by BASEPRI they will
fail.

Zephyr applications will enable IRQs before jumping to main. So I do not see why this is a problem.

File: boot/zephyr/main.c (hunk 1 above)
Comment (sigvartmh) [reply boot/zephyr/main.c:205]: Miss aligned the #else will fix it.
